### PR TITLE
Spectator mode

### PR DESCRIPTION
**Pull request by Nowp**

### CODE REVIEW

File: code/Player/Cameras/SpectatorCamera.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            else if (Input.Pressed(InputButton.Attack2))
+            {
+                playerIndex = (playerIndex + 1) % playerCount;
+            }
+
+            if (0 <= playerIndex && playerIndex < playerCount) 
+                Target = Gamemode.Instance.RoundManager.CurrentRound.Players[playerIndex];
+
+            FieldOfView = SpectatorFieldOfView;
+        }
```
Comment (Prakkkmak) [inline code/Player/Cameras/SpectatorCamera.cs:47]: To method + documentation
Single respon principle <

File: code/Modules/Items/Carriables/Item.cs [hunk 2]
```diff
@@ -60,17 +61,11 @@ public override string ToString()
...
-            return Data.Id == other.Data.Id;
-        }
-
         public override bool Equals(object obj)
         {
             if (ReferenceEquals(null, obj)) return false;
             if (ReferenceEquals(this, obj)) return true;
-            if (obj.GetType() != GetType()) return false;
-            return Equals((Item)obj);
+            return false;
```
Comment (Prakkkmak) [inline code/Modules/Items/Carriables/Item.cs:68]: delete equals

File: code/Player/Cameras/SpectatorCamera.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+namespace ssl.Player.Cameras
+{
+    public partial class SpectatorCamera : Camera
+    {
+        private const float SpectatorFieldOfView = 80F;
+        private const float FocusDistance = 100F;
+        private const float MaxSpeed = 75F;
+        private const float SpeedChangeFactor = 3F;
+
+        private bool clearTargetPressed;
```
Comment (Prakkkmak) [inline code/Player/Cameras/SpectatorCamera.cs:14]: isClear..

File: code/Player/Cameras/SpectatorCamera.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            if (Target != null)
+            {
+                Pos = Target.Position + Input.Rotation.Backward * FocusDistance;
+                Rot = Input.Rotation;
+
+                if (Input.Pressed(InputButton.Jump))
+                {
+                    Target = null;
+                    clearTargetPressed = true;
+                }
```
Comment (Prakkkmak) [inline code/Player/Cameras/SpectatorCamera.cs:41]: external method

File: code/Player/Cameras/SpectatorCamera.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                
+                MoveSpeed += Input.MouseWheel * SpeedChangeFactor;
+                MoveSpeed = MoveSpeed.Clamp(0, MaxSpeed);
+                
+                Vector3 wishDir = new Vector3(Input.Forward, Input.Left, 0).Normal * Rot;
+                wishDir += Vector3.Up * up;
+                
+                Pos += wishDir * MoveSpeed;
+                Rot = Input.Rotation;
```
Comment (Prakkkmak) [inline code/Player/Cameras/SpectatorCamera.cs:70]: External ethod

File: code/Player/Cameras/SpectatorCamera.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            }
+
+            if (Input.Pressed(InputButton.Attack1))
+            {
+                PreviousTarget();
+            }
+            else if (Input.Pressed(InputButton.Attack2))
+            {
+                NextTarget();
+            }
```
Comment (Prakkkmak) [inline code/Player/Cameras/SpectatorCamera.cs:80]: 👍

**Review by Prakkkmak — CHANGES_REQUESTED** [review]

File: code/Modules/Items/Carriables/Item.cs (hunk 2 above)
Comment (Nowp) [reply code/Modules/Items/Carriables/Item.cs:68]: in trashbag branch

File: code/Player/Cameras/SpectatorCamera.cs (hunk 3 above)
Comment (Nowp) [reply code/Player/Cameras/SpectatorCamera.cs:14]: This is to check if the button to clear the target is being pressed

**Review by Prakkkmak — APPROVED** [review]

**Pull request merged**